src/Utilities/StdHelpers.hpp

```diff
@@ -283,3 +283,61 @@ std::string formatted_string(const std::string& fmt, Args... args) {
   }
   return rtn;
 }
+
+// Arithmetic operators for std::array<T, Dim>
+
+template <size_t Dim, typename T>
+inline void operator+=(std::array<T, Dim>& lhs, const std::array<T, Dim>& rhs) {
+  for (size_t i = 0; i < Dim; ++i) {
+    lhs[i] += rhs[i];
+  }
+}
+
+template <size_t Dim, typename T>
+inline std::array<T, Dim> operator+(const std::array<T, Dim>& lhs,
```

Comment:
Should compute `noexcept`ness

Reply:
Here the `noexcept` check I think would be `noexcept(noexcept(std::array<T, Dim> result = lhs && result += rhs))`, if not it might be necessary to use `std::declval<std::array<T, Dim>>{} += rhs`.

Reply:
We'll need to experiment a bit with computing `noexcept`ness of functions with multiple statements, I've only had to do it for single-statement functions so far.

Reply:
I can't get either of these suggestions to compile. I think the reason the second suggestion won't work is that += is not a member function of array, and calling a non-member function which takes a non-const reference doesn't seem to work with declval. I'm not sure what a solution is.

Reply:
```cpp
#include <array>

template <size_t Dim, typename T>
inline void operator+=(std::array<T, Dim>& lhs,
                       const std::array<T, Dim>& rhs) noexcept(noexcept(lhs[0] += rhs[0])) {
  for (size_t i = 0; i < Dim; ++i) {
    lhs[i] += rhs[i];
  }
}

template <size_t Dim, typename T>
inline std::array<T, Dim> operator+(
    const std::array<T, Dim>& lhs,
    const std::array<T, Dim>&
        rhs) noexcept(noexcept(const_cast<std::array<T, Dim>&>(lhs) += rhs)) {
  std::array<T, Dim> result = lhs;
  result += rhs;
  return result;
}

int main() {
  std::array<int, 2> t0{{1, 2}}, t1{{3, 4}};
  std::array<int, 0> t2{{}}, t3{{}};
  t0 + t1;
  static_assert(noexcept(t2 + t3), "");  
  return 0;
}
```

Reply:
You'll still need to check that the creation of `result` is `noexcept`